oxlate/headers.py

```diff
@@ -0,0 +1,38 @@
+from .utils import string_to_case_number, case_number_to_string
+
+class Headers:
+    def __init__(self):
+        self._data = {}
+
+    def add(self, name, value, adjust_case_to_allow_duplicates=False):
+        # import ipdb; ipdb.set_trace()
+        if adjust_case_to_allow_duplicates:
+            if self._data.get(name):
+                # Collision, need to change case
+                same_names_case_insensitive = list(filter(lambda key: key.lower() == name.lower(), self._data.keys()))
+                existing_names_case_binary_representations = set(map(string_to_case_number, same_names_case_insensitive))
```

Comment:
I found `case_number` to be a bit confusing - perhaps `case_indicator`?  Open to other suggestions...